### DISCUSSION TITLE
How can a workflow be triggered on a commit to a branch by a bot ?

### DISCUSSION THREAD

**Discussion by seymurcrisp** [community/community#40808] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm working on a Shopify store and there's a Shopify app to integrate code in the repository with the theme in Shopify Admin. When a change is made in repo it's reflected in the theme in Shopify admin and vice versa. So when the opposite happens (i.e. a change is made in Shopify admin) the Shopify bot commits those changes to the connected branch.

My question is, how can I write a workflow that will be triggered when that bot commits something to the branch. As far as I know, currently there's not an "on commit" option to trigger a workflow.

**Comment by airtower-luna** [comment, marked as answer]
> As far as I know, currently there's not an "on commit" option to trigger a workflow.

The event you're looking for [is called `push`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push). Any change to the HEAD of a branch is a push, whether you do it with `git push` or some other means. :slightly_smiling_face:

**Reply by seymurcrisp** [reply]
Great! Thank you!
